app/upgrades/v16/upgrades.go

```diff
@@ -185,7 +185,7 @@ func CreateUpgradeHandler(
 
 		for _, twapRecord := range clPoolTwapRecords {
 			twapRecord.LastErrorTime = time.Time{}
-			keepers.TwapKeeper.StoreNewRecord(ctx, twapRecord)
+			keepers.TwapKeeper.StoreNewRecord(ctx, twapRecord.Asset0Denom, twapRecord.Asset1Denom, twapRecord)
```

Comment:
We need to be careful about this. We need to ensure that this is storing the record **incorrectly**. If its storing it correctly, then anyone syncing a node from v15 to v16 will have the wrong state, since v16 will store it correctly and then v17 will flip the record and make it incorrect

Reply:
i'm pretty confident that this is set **incorrectly** and we correct it in v17 upgrade handler. This is how it was set in v16 as well and in v17 we reference this record but flip denom0 and denom1

Reply:
yup it is as it nothing's changed